### PR TITLE
Reflect the DEStats changes

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -343,15 +343,15 @@ function solve!(integrator::DDEIntegrator)
             typeof(prob),typeof(integrator.sol.alg),typeof(interp)}(
                 sol_array.u, u_analytic, errors, sol_array.t, interp.ks,
                 prob, integrator.sol.alg, interp, interp.alg_choice,
-                interp.dense, integrator.sol.tslocation, :Success)
+                interp.dense, integrator.sol.tslocation, integrator.integrator.sol.destats, :Success)
```
Comment (ChrisRackauckas) [inline src/solve.jl]: why from integrator.integrator?

Comment (YingboMa) [reply src/solve.jl]: `integrator.sol.destats` gives 0 for some reason.

Comment (ChrisRackauckas) [reply src/solve.jl]: isn't `integrator` being stepped, and `integrator.integrator` just being a store? @devmotion

Comment (devmotion) [reply src/solve.jl]: Yes, `integrator` is stepped but the solution in `integrator.integrator` is used for inter- and extrapolation and updated in https://github.com/JuliaDiffEq/DelayDiffEq.jl/blob/master/src/integrator_interface.jl#L25. Shouldn't you get `integrator.sol.destats == integrator.integrator.sol.destats` if you specify `destats` explicitly in https://github.com/JuliaDiffEq/DelayDiffEq.jl/blob/master/src/solve.jl#L72-L80?

Comment (ChrisRackauckas) [reply src/solve.jl]: Yes, that would do it. I agree that the correct thing to do would be to alias the destats here

**Comment by ChrisRackauckas** [comment]
This passed locally, so I'm at least going to merge and tag since right now everything is broken. I think it just needed the nonlinear solver changes.

**Pull request merged**